src/test/java/org/sagebionetworks/bridge/exporter/integration/ExportTest.java

```diff
@@ -0,0 +1,813 @@
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
+
+        // check again
+        assertExport(null, ExportType.HOURLY, uploadId2, 1, false, true, false);
+    }
+
+    @Test
+    public void testEndDateTimeBeforeLastExportDateTime() throws Exception {
+        DateTime dateTimeBeforeExport = DateTime.now();
+        BridgeExporterRequest exporterRequest = new BridgeExporterRequest.Builder().withExportType(ExportType.HOURLY)
+                .withEndDateTime(dateTimeBeforeExport).withStudyWhitelist(ImmutableSet.of(studyId)).build();
+
+        // the first one should be normal
+        assertExport(exporterRequest, null, uploadId, 0, false, false, false);
+
+        // the second one will not export anything -- synapse table will remain the same
+        assertExport(exporterRequest, null, uploadId, 0, false, false, false);
+
+        // then test endDateTime before lastExportDateTime -- there should be no change as well
+        exporterRequest = new BridgeExporterRequest.Builder().withExportType(ExportType.HOURLY)
+                .withEndDateTime(dateTimeBeforeExport.minusHours(1)).withStudyWhitelist(ImmutableSet.of(studyId)).build();
+
+        sqsHelper.sendMessageAsJson(EXPORTER_SQS_URL, exporterRequest, 5);
+
+        TimeUnit.SECONDS.sleep(30);
+
+        Item lastExportDateTime = ddbExportTimeTable.getItem("studyId", studyId);
+        Long lastExportDateTimeEpoch = lastExportDateTime.getLong("lastExportDateTime");
+        assertNotNull(lastExportDateTimeEpoch);
+
+        assertTrue(lastExportDateTimeEpoch > exporterRequest.getEndDateTime().getMillis());
+    }
+
+    @Test
+    public void testS3Override() throws Exception {
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+
+        // first upload a test override file into s3 bucket
+        s3FileName = "integ-test-redrive-record-ids." + DateTime.now().withZone(DateTimeZone.UTC).toString();
+        List<String> recordIds = ImmutableList.of(uploadStatus.getRecord().getId());
+        s3Helper.writeLinesToS3(recordIdOverrideBucket, s3FileName, recordIds);
+
+        DateTime dateTimeBeforeExport = DateTime.now();
+        LOG.info("Time before request daily exporting: " + dateTimeBeforeExport.toString());
+
+        // then upload another test file
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted");
+
+        BridgeExporterRequest exporterRequest = new BridgeExporterRequest.Builder().withEndDateTime(dateTimeBeforeExport)
+                .withRecordIdS3Override(s3FileName).build();
+
+        // assert only the first upload was exported
+        assertExport(exporterRequest, null, uploadId, 0, false, true, false);
```

Comment:
its already tested: in the assertExport method, i verify if the synapse table row number is equal to the offset + 1 -- which means, in this case, it verifies exporter only export the right one and does not export wrong one.